gobblin-service/src/main/java/gobblin/service/modules/orchestration/Orchestrator.java

```diff
@@ -169,45 +116,59 @@ public void onDeleteSpec(URI deletedSpecURI, String deletedSpecVersion) {
 
     try {
       if (topologyCatalog.isPresent()) {
-        topologyCatalog.get().getSpec(deletedSpecURI);
+        topologyCatalog.get().remove(deletedSpecURI);
         this.specCompiler.onDeleteSpec(deletedSpecURI, deletedSpecVersion);
-        // TODO: Re-compile Flows. Since Topologies are immutable at the moment, we are not handling it.
-      }
-    } catch (SpecNotFoundException e) {
-      _log.info(String.format("Spec with URI: %s was not found in TopologyCatalog, going to try FlowCatalog",
-          deletedSpecURI));
-    }
-
-    try {
-      if (flowCatalog.isPresent()) {
-        flowCatalog.get().getSpec(deletedSpecURI);
-        // If deleted Spec is of type Flow, remove it from all SpecExecutorInstances
-        // (SpecExecutorInstances are expected to no-op if they do not have the URI scheduled on them)
-        if (topologyCatalog.isPresent()) {
-          Collection<TopologySpec> specs = specCompiler.getTopologySpecMap().values();
-          for (TopologySpec topologySpec : specs) {
-            try {
-              // Note: At the moment, the URI of Flow and Job is expected to remain the same.
-              // .. We will need to handle 1 Flow URI to n Job URI compilation when we introduce it.
-              // .. Hence at the moment Flow URI is being deleted on all SpecExecutorInstances.
-              topologySpec.getSpecExecutorInstanceProducer().deleteSpec(deletedSpecURI);
-            } catch (RuntimeException e) {
-              _log.warn(String.format("Did not delete Spec with URI: %s and Version: %s on SpecExecutorInstance: %s",
-                  deletedSpecURI, deletedSpecVersion, topologySpec.getSpecExecutorInstanceProducer().getUri()), e);
-            }
-          }
-        }
       }
-    } catch (SpecNotFoundException e) {
-      _log.info(String.format("Spec with URI: %s was not found in TopologyCatalog, going to try FlowCatalog",
-          deletedSpecURI));
+    } catch (RuntimeException e) {
```

Comment:
Removed. Earlier Orchestrator was listener to both Flow and Topology Catalog creating a situation where it cannot distinguish between a Flow and Topology URI - hence no way of knowing if missing Spec runtime exception are really an issue or not. 
However, with recent changes Orchestrator only listens to Topology Catalog. So it is not needed anymore.